packages/context/src/binding.ts

```diff
@@ -333,4 +333,39 @@ export class Binding {
     this.isLocked = false;
     return this;
   }
+
+  asNestedProperty(path: string): Binding {
+    const nested = new Binding(this.key, true);
+    nested.getValue = (ctx) => {
+      const value = this.getValue(ctx);
+      if (isPromise(value)) {
+        return value.then((v: BoundValue) => getValue(v, path));
+      } else {
+        return getValue(value, path);
+      }
+    };
+
+    // TODO(bajtos) override value setters to forbid modification
+    // TODO(bajtos) inherit scope, tag, etc.
+
+    return nested;
+  }
+}
+
+/**
+ * Get the value by `.` notation
+ * @param obj The source value
+ * @param path A path to the nested property, such as `x`, `x.y`, `x.length`,
```

Comment:
Missing the type definitions, and the return type:
```ts
/**
 * Get the value by `.` notation
 * @param {Object} obj The source value
 * @param {String} path A path to the nested property, such as `x`, `x.y`, `x.length`,
 * or `x.0`
 * @returns {BoundValue} The BoundValue reference.
 */
```

Reply:
Ah, this was already here before. Up to you if you want to add the changes or leave it for another PR.

Reply:
Type definitions are not needed, tsdoc parser will infer them from the type information provided by TypeScript.